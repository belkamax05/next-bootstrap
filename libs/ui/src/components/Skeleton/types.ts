export interface SkeletonProps {
  animated?: boolean;
  borderRadius?: string;
  count?: number;
  color?: string;
  highlightColor?: string;
  height?: string;
  space?: string;
  width?: string;
  id: string;
}
