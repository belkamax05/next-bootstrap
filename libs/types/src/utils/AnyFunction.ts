/**
 * Replacement for "() => unknown"
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type AnyFunction = (...args: unknown[]) => unknown;
