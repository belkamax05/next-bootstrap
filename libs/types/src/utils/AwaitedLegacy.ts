export type AwaitedLegacy<T> = T extends PromiseLike<infer U> ? U : T;
