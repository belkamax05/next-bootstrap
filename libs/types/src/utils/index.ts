export * from './AnyFunction';
export * from './AnyObject';
export * from './ArrayElement';
export * from './AwaitedLegacy';
export * from './DeepPartial';
export * from './DeepRecord';
export * from './ExcludePartials';
export * from './hooks';
export * from './hot';
export * from './LooseAutocomplete';
export * from './PickPartials';
export * from './PickRequired';
export * from './PromiseOrValue';
export * from './QueryParams';
export * from './QueryParamType';
export * from './ResolveObject';
export * from './SearchQueryRecord';
