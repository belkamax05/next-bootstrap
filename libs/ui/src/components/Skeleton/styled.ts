import excludeProps from '@/utils/helpers/excludeProps';
import { keyframes, styled } from '@mui/material/styles';
import { SkeletonProps } from './types';

interface LineProps extends SkeletonProps {
  animated?: boolean;
}

const shimmer = keyframes`
  0% {
    background-position: 0 0;
  }
  100% {
    background-position: -200% 0;
  }
`;

const name = 'Skeleton';

export const StyledContainer = styled('div', {
  name,
  slot: 'container',
  shouldForwardProp: excludeProps(['width']),
})<{ width?: SkeletonProps['width'] }>(({ width }) => ({
  display: 'flex',
  flexDirection: 'column',
  width: width || '100%',
}));

export const StyledLine = styled('span', {
  name,
  slot: 'line',
  shouldForwardProp: excludeProps([
    'height',
    'borderRadius',
    'color',
    'highlightColor',
    'space',
    'animated',
  ]),
})<LineProps>(
  ({
    height = '100%',
    borderRadius = '0px',
    color = 'rgba(0, 0, 0, 0.1)',
    highlightColor = 'rgba(0, 0, 0, 0.14)',
    space = '10px',
    animated,
  }) => ({
    backgroundImage: `linear-gradient(90deg, ${color} 8%, ${highlightColor} 18%,  ${color} 33%)`,
    borderRadius,
    backgroundSize: '200% 100%',
    display: 'block',
    width: '100%',
    height,
    minHeight: '5px',
    marginBottom: space,
    position: 'relative',
    overflow: 'hidden',
    '&:last-of-type': {
      marginBottom: 0,
    },
    ...(animated && {
      animation: `${shimmer} 1.5s infinite linear`,
      animationTimingFunction: 'linear',
    }),
  }),
);
