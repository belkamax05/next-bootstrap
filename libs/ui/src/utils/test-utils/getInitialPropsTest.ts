import { AnyFunction, DeepPartial, PageContext } from '@/types';

const getInitialPropsTest = (
  func: AnyFunction,
  ctx?: DeepPartial<PageContext>,
  ...rest: unknown[]
) => {
  it('should not fail without props', async () => {
    const result = await func(ctx, ...rest);

    expect(result).toBeTruthy();
    expect(result).toMatchSnapshot();
  });
};

export default getInitialPropsTest;
