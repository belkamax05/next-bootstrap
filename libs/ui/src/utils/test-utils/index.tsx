export { act, fireEvent, render, screen, waitFor, within } from '@testing-library/react';
