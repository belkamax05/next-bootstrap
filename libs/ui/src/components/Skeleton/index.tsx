import { StyledContainer, StyledLine } from './styled';
import { SkeletonProps } from './types';

export const skeletonTestIds = {
  skeleton: 'skeleton',
  skeletonElement: 'skeletonElement',
} as const;

const Skeleton = ({ animated, count = 1, id }: SkeletonProps) => {
  return (
    <StyledContainer data-testid={skeletonTestIds.skeleton}>
      {[...new Array(count)].map((_, index) => (
        <StyledLine
          id={`skeleton_${id}_${index}`}
          animated={animated}
          key={`skeleton_${id}_${index}`}
          data-testid={skeletonTestIds.skeletonElement}
        >
          &zwnj;
        </StyledLine>
      ))}
    </StyledContainer>
  );
};

export default Skeleton;
