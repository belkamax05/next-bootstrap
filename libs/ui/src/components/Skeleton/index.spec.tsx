import { render, screen } from '@/ui/utils/test-utils';
import Skeleton, { skeletonTestIds } from '.';

describe('Skeleton', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Skeleton id="test" />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render 5 loading bars', () => {
    render(<Skeleton count={5} id="test" />);
    const element = screen.queryByTestId(skeletonTestIds.skeleton);
    expect(element.children.length).toBe(5);
  });
});
