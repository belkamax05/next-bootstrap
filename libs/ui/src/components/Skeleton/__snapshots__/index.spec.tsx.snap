// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render successfully 1`] = `
<body>
  <div>
    <div
      class="css-11t9lrj-Skeleton-container"
      data-testid="skeleton"
    >
      <span
        class="css-jjbo1d-Skeleton-line"
        data-testid="skeletonElement"
        id="skeleton_test_0"
      >
        â€Œ
      </span>
    </div>
  </div>
</body>
`;
